Sem. 07/Practicum/Nvector/NVector.cpp

```
@@ -0,0 +1,198 @@
+#include "NVector.h"
+
+Nvector::Nvector()
+{
+	size = 1;
+	coordinates = new double[size];
+	coordinates[0] = 0;
+}
+
+Nvector::Nvector(const double* coordinates, const size_t size)
+{
+	this->size= size;
+	this->coordinates = new double[size];
+	for (int i = 0; i < size; i++) {
+		this->coordinates[i] = coordinates[i];
+	}
+}
+
+Nvector::Nvector(const Nvector& other)
+{
+	copy(other);
+}
+
+Nvector& Nvector::operator=(const Nvector& other)
+{
+	if (this != &other) {
+		free();
+		copy(other);
+	}
+	return *this;
+}
+
+Nvector::~Nvector()
+{
+	free();
+}
+
+Nvector& Nvector::operator+=(const Nvector& other)
+{
+	if (size != other.size) {
+		std::cout<<"Different sizes!"<<std::endl;
```

It's better to throw an exception here or at least to write on the **cerr** output instead of the standard output to log the error.